### PR TITLE
Add signing for macos package

### PR DESCRIPTION
**Pull request by jimboslicethat**
I was thinking perhaps @parkerholladay could look at this one and test the happy path and the unhappy path with ubuntu and mac.

### CODE REVIEW

File: scripts/package-zipper.js [hunk 1]
```diff
@@ -1,52 +1,50 @@
...
 
-function zipPackage(os) {
-  const outputDir = getOutputDir(os)
-  const sourceDir = getSourceDir(os)
+  const srcCodeExists = fs.existsSync(sourceDir)
+  const nonMacMessage = os.type() !== 'Darwin'
 
-  if (!fs.existsSync(sourceDir))
+  if (!srcCodeExists)
     throw new Error('npm run build:packages must be run before zipping!')
```
[the remark this answers is not in the document]
Comment (jimboslicethat) [reply scripts/package-zipper.js]: I dont think so. What did you have in mind?

File: scripts/package-zipper.js [hunk 2]
```diff
@@ -1,52 +1,50 @@
...
-}
+function zipPackage(operatingSystem) {
+  const zippedDir = getPackageZipDir(operatingSystem)
+  const sourceDir = getPackageSrcDir(operatingSystem)
 
-function zipPackage(os) {
-  const outputDir = getOutputDir(os)
-  const sourceDir = getSourceDir(os)
+  const srcCodeExists = fs.existsSync(sourceDir)
+  const nonMacMessage = os.type() !== 'Darwin'
```
[the remark this answers is not in the document]
Comment (jimboslicethat) [reply scripts/package-zipper.js]: Sounds great! 👍

File: scripts/osx-sign.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+function nonMacWarning() {
+  const warningMessage =
+    'Warning: Unable to sign the current macos release.'.concat(
+      '\nThe macos package may only be signed from a macOS.'
+    ).concat(
+      '\nThe existing macos release will now be removed and will not be included when running: npm run release:publish'
+    )
+
+  console.warn(warningMessage)
+  removePackageSrc('macos')
```
[the remark this answers is not in the document]
Comment (jimboslicethat) [reply scripts/osx-sign.js]: Ooooh I see what you're saying.

File: README.md [hunk 4]
```diff
@@ -28,27 +28,37 @@ npm run release:create
...
+```
+brew cask install xquartz
+brew install wine
+```
+
+### **Linux Users**
+* Your release will _**not**_ include a zipped package for macOS.
+* The MacOS package must be signed with the appropriate `Pluralsight Developer Certificate` stored on the users MacOS keychain.
+* Building an Electron app for the Windows target platform requires [Wine 1.6](https://www.winehq.org/) or later to be installed.
+* Directions for installing `wine` on Linux can be found [here](hhttps://www.winehq.org/download)
```
Comment (parkerholladay) [inline README.md]: This link is broken (starts with `hhttps` instead of `https`)

**Pull request merged**